### PR TITLE
Add com.github.corna.Vivado

### PR DESCRIPTION
**Pull request by corna**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

-----

Hi all.

This PR supersedes #2164

This PR adds the [Xilinx Vivado Design Suite](https://www.xilinx.com/downloads) as flatpak package. All the Xilinx material is NOT redistributable and requires a login to be downloaded (and the filling of a "U.S. Government Export Approval" form). For this reason, this repository does not contain any Xilinx code and instead works as a wrapper over an "official" installation. The user is prompted to download the official installation tool or to provide login information to automatically download the installer, which is then passed to this package and installed within the sandbox (in the `$XDG_DATA_HOME/xilinx-install` folder).

This package works (hopefully) with any version of the Xilix Vivado Design Suite. If multiple versions are installed, the user has to select one after clicking on the application icon.

### Permissions
 - `share=ipc`, `socket=wayland` and `socket=x11` are obvious.
 - `share=network` is required for the correct functionality of docnav (which downloads the documentation from the internet) and is required for licensing (as the license is bound to the MAC address). 
 - `device=all` is required for accessing the hardware devices (USB), which can be programmed through Vivado. For this, the user has to install some udev rules, see below.
 - `filesystem=~/.Xilinx:create`: this folder is used by the Xilinx tools to store some data and the license files.
 - `filesystem=xdg-documents` and `filesystem=xdg-desktop`: the user has to store its project somewhere. I don't know however if it is better to provide just the documents and desktop (better sandbox) or the whole home directory (more user-friendly).
 - `persist=.`: not really needed, but it is better to keep the whole home persistent  so that, if the user saves a project outside the Desktop or Documents directory, it is not lost as soon as the application is closed

### Developer contact

Regarding [the requirements][reqs], I have not contacted Xilinx regarding this package (I don't even know where I should contact them, there does not seem to be an email for non-paying customers).

### Icons

The better-suited icons for this package (to be consistent with an official installation) are the Vivado icon, the Vitis icon and the Xilinx logo. These are in the sub-module [com.github.corna.Vivado.icons](https://github.com/corna/com.github.corna.vivado.icons).

### udev rules

The tools can program the boards through many USB-JTAG adapters, which however require an udev rule to correctly set the permissions.

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build Com.GitHub.corna.vivado

**Comment by hfiguiere** [comment]
bot, build com.github.corna.Vivado

**Comment by hfiguiere** [comment]
using the right case.

**Comment by hfiguiere** [comment]
The appdata validation fails:

builddir/files/share/appdata/com.github.corna.Vivado.appdata.xml: FAILED:
• style-invalid         : Too many <p> tags for a good description [14/10]

File: README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# Flatpak package for Xilinx Vivado Design Suite
```
[the remark this answers is not in the document]
Comment (corna) [reply README.md:1]: Yes, I have copied the whole README in the appdata description

**Comment by corna** [comment]
> The appdata validation fails:
> 
> ```
> builddir/files/share/appdata/com.github.corna.Vivado.appdata.xml: FAILED:
> • style-invalid         : Too many <p> tags for a good description [14/10]
> ```

I tried to use lists (and other items like `<em>` and `<code>`, as described [here](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-description)) to separate the FAQs items, however `flatpak-builder` failed the validation of the appdata XML on my PC (Arch Linux). Are those tags (`<ul>`, `<em>` and `<code>`) allowed on flatpak/flathub?

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
This should be complete now and I have been using it for a while without any particular bug.
Let me know if it is missing something.

File: com.github.corna.Vivado.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+runtime: org.gnome.Sdk
+runtime-version: '40'
+sdk: org.gnome.Sdk
+command: vivado
+tags:
+  - proprietary
+
+finish-args:
+  - --share=ipc
+  - --socket=wayland
```
Comment (hfiguiere) [inline com.github.corna.Vivado.yml:11]: does the software even support Wayland?

File: com.github.corna.Vivado.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+app-id: com.github.corna.Vivado
+runtime: org.gnome.Sdk
+runtime-version: '40'
+sdk: org.gnome.Sdk
```
Comment (hfiguiere) [inline com.github.corna.Vivado.yml:4]: does it really need the GNOME runtime?

If all you need is Gtk3 then it's in the freedesktop-sdk

File: com.github.corna.Vivado.yml (hunk 2 above)
Comment (corna) [reply com.github.corna.Vivado.yml:11]: Not yet AFAIK

File: com.github.corna.Vivado.yml (hunk 3 above)
Comment (corna) [reply com.github.corna.Vivado.yml:4]: I used freedesktop.Sdk previously, however Vitis (eclipse-based) fails to start with the freedesktop one on X11 (see commit 7069d96e1cfa0c3a6bfde8ef5d8d77819a20998a). I haven't yet understood why though.

Comment (hfiguiere) [reply com.github.corna.Vivado.yml:4]: ok

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

File: resources/share/metainfo/com.github.corna.Vivado.appdata.xml [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- Copyright 2021 Nicola Corna <nicola@corna.info> -->
+<component type="desktop">
+  <id>com.github.corna.Vivado</id>
+  <metadata_license>CC0-1.0</metadata_license>
+  <name>Xilinx Vivado Design Suite</name>
+  <summary>Xilinx Design Suite for hardware development</summary>
```
Comment (barthalion) [inline resources/share/metainfo/com.github.corna.Vivado.appdata.xml:7]: I think it needs to be clarified it's not official Xilinx app, at best an installer helper. It should also carry a disclaimer like https://flathub.org/apps/details/com.jetbrains.GoLand does in `description`

Comment (corna) [reply resources/share/metainfo/com.github.corna.Vivado.appdata.xml:7]: It is already clarified in the description at line 9 ("This package provides a flatpak wrapper over an official installation. As the Xilinx tools are not redistributable, this package does not contain any Xilinx-licensed material, which is instead downloaded on-the-fly through the official Xilinx installer.")

I have also added the disclaimer, as you suggested.

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
bot, build com.github.corna.Vivado

**Comment by corna** [comment]
With 487c7e6 the user experience should be basically the same as other apps (like Steam): on the first run, the software is automatically downloaded (after a login) and installed.

**Comment by scraich95** [comment]
I've been using this package for a while and I'm very happy with it, will it be merged any soon?

**Comment by nedrichards** [comment]
bot, build com.github.corna.Vivado

**Pull request closed without merging**